corehq/apps/hqcase/api/updates.py

```diff
@@ -306,15 +305,18 @@ def _get_owner_ids(domain, data, case_db):
     owner_ids = []
     case_ids = []
     for data_item in data:
-        if data_item.is_new_case:
+        if data_item.owner_id:
             owner_ids.append(data_item.owner_id)
-        else:
+        try:
             case_id = data_item.get_case_id(case_db)
             case_ids.append(case_id)
+        except UserError as e:
```

Comment:
the logic here is a little weird - if `is_new_case`, then `get_case_id` is a UUID generated here, so it doesn't really make sense to check that it's valid.
https://github.com/dimagi/commcare-hq/blob/0a602cdc5d324507a3b5040d6309d9b80a5af3c2/corehq/apps/hqcase/api/updates.py#L129-L129

If it's an update, the `UserError` raised here means we can't tell what case they're trying to update, which is a hard failure regardless of whether `owner_id` is specified.  I think you can just let that exception be raised.